library/src/scala/tasty/util/ShowSourceCode.scala

```diff
@@ -235,31 +275,22 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
             this += " => "
             printTree(rhs)
             this += ")"
-
-          case Term.Apply(Term.Ident("while$"), _) =>
-            val DefDef("while$", _, _, _, Some(Term.If(cond, Term.Block(body :: Nil, _), _))) = stats.head
-            this += "while ("
-            printTree(cond)
-            this += ") "
-            printTree(body)
-
-          case Term.Apply(Term.Ident("doWhile$"), _) =>
-            val DefDef("doWhile$", _, _, _, Some(Term.Block(List(body), Term.If(cond, _, _)))) = stats.head
-            this += "do "
-            printTree(body)
-            this += " while ("
-            printTree(cond)
-            this += ")"
-
+          case expr if isLoopEntryPoint(expr) && stats.size == 1 =>
```

Comment:
I think it would be cleaner to have this case be part of the extractor:
```scala
def unapply(arg: Tree)(implicit ctx: Context): Option[(Term, List[Statement])] = arg match {
  case DefDef("while$", _, _, _, Some(Term.If(cond, Term.Block(bodyStats, _), _))) => Some((cond, bodyStats))
  case Block(List(tree), _) => unapply(tree)
  case _ => None
}
```